src/vm/codeversion.cpp

```diff
@@ -2157,6 +2157,8 @@ PCODE CodeVersionManager::PublishVersionableCodeIfNecessary(MethodDesc* pMethodD
             pCode = pMethodDesc->PrepareCode(activeVersion);
         }
 
+        MethodDescBackpatchInfoTracker::ConditionalLockHolder lockHolder;
```

Comment:
Could move this to just before where the `TableLockHolder` is instantiated, and pass in `pMethodDesc->MayHaveEntryPointSlotsToBackpatch()` as the argument, and similarly in other new places where this lock is taken

Reply:
See other comments about lock order and thread suspension.